pages/common/csc.md

```diff
@@ -0,0 +1,35 @@
+# csc
+
+> The Microsoft C# Compiler.
+
+- Compile one or more C# files to a CIL executable:
+
+`csc {{path/to/input_file_a.cs}} {{path/to/input_file_b.cs}}`
+
+- Specify the output filename:
+
+`csc /out:{{path/to/filename}} {{path/to/input_file.cs}}`
+
+- Compile into a '.dll' library instead of an executable:
+
+`csc /target:library {{path/to/input_file.cs}}`
+
+- Reference another assembly:
+
+`csc /reference:{{path/to/library.dll}} {{path/to/input_file.cs}}`
+
+- Embed a resource:
+
+`csc /resource:{{path/to/input_file.cs}},{{namespace.string.name}}`
+
+- Automatically generate XML documentation:
+
+`csc /doc:{{path/to/output.xml}} {{path/to/input_file.cs}}`
+
+- Specify an icon:
+
+`csc /win32icon:{{path/to/icon.ico}} {{path/to/input_file.cs}}`
+
+- Strongly-name the resulting assembly with a keyfile:
```

Comment:
What does `Strongly-name` mean here? 🙂

Reply:
Strong-naming is a concept specific to .NET assemblies in which the IL in an assembly is hashed, and then the hash encrypted with a private key. The public key is then bundled along with the assembly. In this manner, it can be assured that the code inside a .NET assembly has not been altered since it was built.

Combined with an install-time verification system such as `signtool.exe` (Code-Signing certificates; Windows) or GPG in a package manager (Linux), a full chain of trust can be established to ensure that the executing IL has both not been altered since build time, and was built by a trusted party.

Reply:
Cheers for the explanation. :smile:

Reply:
np!